core/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -234,32 +270,38 @@ default void restore() {
     }
 
     static final class ThreadContextStruct {
-        private final Map<String,String> headers;
+        private final Map<String, String> persistentHeaders;
```

Comment:
can this be ```requestHeaders```? I find it a bit confusing as response headers should also be persistent